zebrad/src/components/tokio.rs

```diff
@@ -23,3 +25,34 @@ impl TokioComponent {
         })
     }
 }
+
+/// Zebrad's handler for various signals
+async fn shutdown() -> Result<(), Report> {
```

Comment:
Issue: let's document that this function only handles terminal interrupts.

(The ctrl_c function is named after the typical unix shell interrupt keystroke, but there are multiple ways to trigger the underlying signal.)

As an alternative, we could add support for `SIGTERM`. `SIGTERM` is the standard unix process termination signal, used by timers and launchers (like systemd). If we support `SIGTERM`, developers will be able to do timed or launched flamegraphs. (If we don't, they'll have to configure a specific termination signal. But some timers and launchers aren't configurable.)

Note that linkerd handles `SIGTERM` for similar reasons:
https://github.com/linkerd/linkerd2-proxy/blob/1b3e899efad2ad4ee471c5b49482437e7a325bde/linkerd/signal/src/lib.rs#L19

I would really like us to support `SIGTERM`, but I don't want to block the PR on that feature.

But I do think it's really important to document the current behaviour.

```suggestion
/// Zebrad's terminal interrupt handler.
///
/// Responds to SIGINT (typically Ctrl-C) on Unix, and Ctrl-C on Windows.
async fn terminal_interrupt() -> Result<(), Report> {
```

Reply:
I think the SIGTERM handling is justified for the reasons you mentioned.

Reply:
Fixed now, though I ended up keeping the function called `shutdown`. I think it's a better description of the function's role, which is to block the application to keep the main thread alive until the user decides to explicitly request a clean shutdown.

Reply:
SIGTERM isn't usually a terminal interrupt, so shutdown is a good name for a function that handles multiple signals.